Makefile

```diff
@@ -1,321 +1,52 @@
-	docker run --privileged \
-		-v /var/run/docker.sock:/var/run/docker.sock  \
-		-e GOPATH=${GOPATH} \
-		-it --entrypoint /bin/bash --rm \
-		$(BUILD_IMAGE) -c \
-		"/usr/local/bin/kind create cluster --loglevel debug --name ${KIND_CLUSTER} --wait 60s ${KIND_CONFIG} --image $(BUILD_IMAGE)"
-
-	#kind create cluster --loglevel debug --name ${KIND_CLUSTER} --wait 60s ${KIND_CONFIG} --image $(BUILD_IMAGE)
-
-${TMPDIR}:
-	mkdir -p ${TMPDIR}
-
-maybe-prepare:
-ifeq ($(SKIP_KIND_SETUP), 0)
-	$(MAKE) prepare
-endif
-
-clean:
-	docker run --privileged \
-		-v /var/run/docker.sock:/var/run/docker.sock  \
-		-e GOPATH=${GOPATH} \
-		-it --entrypoint /bin/bash --rm \
-		$(BUILD_IMAGE) -c \
-		"/usr/local/bin/kind delete cluster --name ${KIND_CLUSTER}" 2>&1 || true
-	#kind delete cluster --name ${KIND_CLUSTER} 2>&1 || true
-
-maybe-clean:
-ifeq ($(SKIP_CLEANUP), 0)
-	$(MAKE) clean
-endif
-
-# Generate junit reports, upload to testgrid, fail if conditions are met.
-# Failure is based on test status - may exclude some tests.
-report:
-	bin/report.sh
-
-info:
-	# Get info about env, for debugging
-	set -x
-	pwd
-	env
-	id
-
-
-# Copy source code from the current machine to the docker.
-sync:
-ifneq ($(MOUNT), 1)
-	docker exec ${KIND_CLUSTER}-control-plane mkdir -p ${GOPATH}/src/istio.io/installer \
-		${GOPATH}/src/istio.io
-	docker cp . ${KIND_CLUSTER}-control-plane:${GOPATH}/src/istio.io/installer
+# WARNING: DO NOT EDIT, THIS FILE IS PROBABLY A COPY
+#
+# The original version of this file is located in the https://github.com/istio/common-files repo.
+# If you're looking at this file in a different repo and want to make a change, please go to the
+# common-files repo, make the change there and check it in. Then come back to this repo and run
+# "make updatecommon".
+
+# Copyright 2019 Istio Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+RUN =
+
+# Set the enviornment variable BUILD_WITH_CONTAINER to use a container
+# to build the repo. The only dependencies in this mode are to have make and
+# docker. If you'd rather build with a local tool chain instead, you'll need to
+# figure out all the tools you need in your environment to make that work.
+export BUILD_WITH_CONTAINER ?= 0
+ifeq ($(BUILD_WITH_CONTAINER),1)
+IMG = gcr.io/istio-testing/build-tools:2019-08-05
+UID = $(shell id -u)
+PWD = $(shell pwd)
+GOBIN_SOURCE ?= $(GOPATH)/bin
+GOBIN ?= /work/out/bin
+
+RUN = docker run -t --sig-proxy=true -u $(UID) --rm \
```

Comment:
I don't think it'll work for kind - it creates files under /home/USER/.kube (the kubeconfigs to be used later steps )